### ISSUE TITLE
[Feature Request] Auto-Nested Bullets

### ISSUE THREAD

**Issue by alichtman** [dkarter/bullets.vim#33]
This is an awesome plugin! I use it all the time for note-taking.

One style of note taking that I use a lot is:

```
- Top level
    * Sub point
        + Sub-sub point
```

It'd be great if, after starting a new bullet line, I could hit `TAB` to automatically indent the bullet point and change it to the next "sub-level type" of bullet. Additionally, I'd like to be able to `Shift + TAB` back to the original level of indentation and correct "bullet type". I know that these key mappings may not be what's wanted by the general community, but if they were remappable, that'd be awesome.

I don't have any free time to look into this at the moment, but in a few months, I may be able to come back to it. If you have any suggestions for implementing this, please leave them below.

**Comment by dkarter** [comment]
Hi again @alichtman this is a great suggestion!

I currently use `C-d` and `C-t` in INSERT mode to indent / de-indent a bullet.

To avoid conflicts with auto-completion plugins and snippet expanders who are already competing for the highly coveted `TAB` key it would be best if we instead utilize `C-d` and `C-t` and as in the case of the `<CR>` mapping we already use here for adding a bullet send the original user intent if they are not within the context of a bullet.

I'll be happy to review a PR for this and help guide you if you need help.

**Comment by alichtman** [comment]
From a quick glance at the source code, my general plan to implement this would be something along the lines of:

```vimscript
call s:add_local_mapping('nnoremap', 'C-t', ':call <SID>indent-bullet-level()<cr>')
call s:add_local_mapping('nnoremap', 'C-d', ':call <SID>unindent-bullet-level()<cr>')
```

and in each of those two functions, I'd figure out where in the `[ +, *, - ]` sequence the line is, and indent / deindent, and then replace the bullet character with the next / previous one.

**Comment by alichtman** [comment]
Also, can you assign this issue to me?

**Comment by alichtman** [comment]
Well, I went from knowing nothing about vimL to knowing something about it, but I don't have the time / vimL skills currently to write this feature. I've got a very broken branch that I'll save to continue working on when I do...

**Comment by dkarter** [comment]
Implemented in #54, thanks to @kaymmm !
